scico/test/linop/test_diag.py

```diff
@@ -187,3 +187,146 @@ def test_norm_except(self):
+        )
+        n1 = Id.norm(ord=ord)
+        n2 = D.norm(ord=ord)
+        snp.testing.assert_allclose(n1, n2, rtol=1e-6)
+
+    def test_norm_except(self):
+        input_shape = (5,)
+
+        Id = linop.Identity(input_shape=input_shape, input_dtype=np.float32)
+        with pytest.raises(ValueError):
+            n = Id.norm(ord=3)
+
+
+class TestIdentity:
+    def setup_method(self, method):
+        self.key = jax.random.PRNGKey(12345)
+
+    input_shapes = [(8,), (8, 12), ((3,), (4, 5))]
+
+    @pytest.mark.parametrize("input_dtype", [np.float32, np.complex64])
+    @pytest.mark.parametrize("input_shape", input_shapes)
+    def test_eval(self, input_shape, input_dtype):
+        x, key = randn(input_shape, dtype=input_dtype, key=self.key)
+
+        Id = linop.Identity(input_shape=input_shape, input_dtype=input_dtype)
+        assert (Id @ x).shape == Id.output_shape
+        snp.testing.assert_allclose(x, Id @ x, rtol=1e-5)
+
+    @pytest.mark.parametrize("operator", [op.add, op.sub])
+    @pytest.mark.parametrize("input_shape", input_shapes)
+    def test_binary_op(self, input_shape, operator):
+        input_dtype = np.float32
+        diagonal, key = randn(input_shape, dtype=input_dtype, key=self.key)
+        scalar, key = randn((), dtype=input_dtype, key=key)
+        x, key = randn(input_shape, dtype=input_dtype, key=key)
+
+        Id = linop.Identity(input_shape=input_shape)
+        Ids = linop.ScaledIdentity(scalar=scalar, input_shape=input_shape)
+        D = linop.Diagonal(diagonal=diagonal)
+
+        IdD = operator(Id, D)
+        assert isinstance(IdD, linop.Diagonal)
+        snp.testing.assert_allclose(IdD @ x, operator(1.0, diagonal) * x, rtol=1e-6)
+
+        DId = operator(D, Id)
+        assert isinstance(DId, linop.Diagonal)
+        snp.testing.assert_allclose(DId @ x, operator(diagonal, 1.0) * x, rtol=1e-6)
+
+        IdIds = operator(Id, Ids)
+        assert isinstance(IdIds, linop.ScaledIdentity)
+        snp.testing.assert_allclose(IdIds @ x, operator(1.0, scalar) * x, rtol=1e-6)
+
+        IdsId = operator(Ids, Id)
+        assert isinstance(IdsId, linop.ScaledIdentity)
+        snp.testing.assert_allclose(IdsId @ x, operator(scalar, 1.0) * x, rtol=1e-6)
+
+    def test_scale(self):
+        input_shape = (5,)
+        input_dtype = np.float32
+        scalar, key = randn((), dtype=input_dtype, key=self.key)
+        x, key = randn(input_shape, dtype=input_dtype, key=key)
+        Id = linop.Identity(input_shape=input_shape, input_dtype=input_dtype)
+
+        sId = scalar * Id
+        assert isinstance(sId, linop.ScaledIdentity)
+        snp.testing.assert_allclose(sId @ x, scalar * x, rtol=1e-6)
+
+        Ids = Id * scalar
+        assert isinstance(Ids, linop.ScaledIdentity)
+        snp.testing.assert_allclose(Ids @ x, scalar * x, rtol=1e-6)
+
+        Idds = Id / scalar
+        assert isinstance(Idds, linop.ScaledIdentity)
+        snp.testing.assert_allclose(Idds @ x, x / scalar, rtol=1e-6)
```

Comment:
Another possible issue to create: can we have a standard function to test scaling (and other similar properties) rather than copy/pasting variants of this?

Reply:
One of many aspects of the tests that could be handled better. But this one shouldn't be terrible to fix. Please make an issue.